poll/poll.py

```diff
@@ -140,53 +142,60 @@ def gather_items(self, data, result, noun, field, image=True):
             source_items = []
             result['success'] = False
             result['errors'].append(
-                "'{0}' is not present, or not a JSON array.".format(field))
+                self.ugettext("'{0}' is not present, or not a JSON array.").format(field))
         else:
             source_items = data[field]
 
         # Make sure all components are present and clean them.
         for item in source_items:
             if not isinstance(item, dict):
                 result['success'] = False
+                # Translators: {noun} is either "Answer" or "Question". {item} identifies the answer or question.
                 result['errors'].append(
-                    "{0} {1} not a javascript object!".format(noun, item))
+                    self.ugettext("{noun} {item} not a javascript object!").format(noun=noun, item=item))
```

Comment:
@asadiqbal08 Thanks for updating the strings to use named placeholders, and for adding translator comments. 

You'll need to make some adjustments to the translator comments: Based on the info in [this](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/conventions/internationalization/i18n.html#python-source-code) section of edX's i18n guidelines, translator comments need to be located right above the strings that they comment on. So the three lines above need to be changed to:

```python
                result['errors'].append(
                    self.ugettext(
                        # Translators: {noun} is either "Answer" or "Question". {item} identifies the answer or question.
                        "{noun} {item} not a javascript object!"
                    ).format(noun=noun, item=item))
```

You could also do this (now that I see both of these approaches next to each other, I think I'd prefer the second one; it looks a lot cleaner):

```python
                # Translators: {noun} is either "Answer" or "Question". {item} identifies the answer or question.
                error_message = self.ugettext("{noun} {item} not a javascript object!").format(noun=noun, item=item)
                result['errors'].append(error_message)
```

The same comment applies to the remaining translator comments in this file.